File: frontend/gamingDate/lib/model/friendgroup.dart
import 'package:gaming_date/model/game.dart';
import 'package:gaming_date/model/profile.dart';

class FriendGroup {
  String _friendGroupId;
  String _friendGroupName;
  Game _lastFavouriteGame;
  List<Profile> _members;

  FriendGroup(this._friendGroupId, this._friendGroupName,
      this._lastFavouriteGame, this._members);

  String get friendGroupId => _friendGroupId;

  List<Profile> get members => _members;

  Game get lastFavouriteGame => _lastFavouriteGame;

  String get friendGroupName => _friendGroupName;
}

File: backend/gamingdate/src/main/java/com/gamingdate/model/FriendGroup.java
package com.gamingdate.model;

import java.util.List;

import lombok.Data;
import lombok.NonNull;

@Data
public class FriendGroup {
    @NonNull String friendGroupId;
    @NonNull List<Profile> friendProfiles;
    @NonNull String friendGroupName;
    String lastFavouriteGame;
}
